### ISSUE TITLE
Rule suggestion (improvement): curly should enforce braces/no-braces for all blocks

### ISSUE THREAD

**Issue by lonix1** [palantir/tslint#4477]
### Rule Suggestion
general problem

**What does your suggested rule do?**

The `curly` rule is almost perfect. I'm using `"curly": [true, "ignore-same-line", "as-needed"]`, which allows:
````js
if (foo) {
  bar();
}
else {
  baz();
}
````
and
````js
if (foo)
  bar();
else
  baz();
````

But what if one of the blocks is 1 line and the other block is n lines? Then the safe/correct way is to use braces for both like this:
````js
if (foo) {
  bar();
  qux();
}
else {
  baz();
}
````

However that isn't allowed. What tsline enforces is this unsafe format:
````js
if (foo) {
  bar();
  qux();
}
else               // ---- bad idea for one block to be braced and other not!
  baz();
````

It would be nice to have another option called `"balanced-braces"` or something like that. It ensures that, regardless of the other options, when one block is braced then all must be braced.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
